logical/system_config.go

```diff
@@ -0,0 +1,16 @@
+package logical
+
+import "time"
+
+// SystemConfig exposes system configuration information in a safe way
+// for other logical backends to consume
+type SystemConfig struct {
```

Comment:
This should be an interface instead of a concrete type most likely. Seems odd to have it hold function pointers like this.